.github/workflows/gradle-check.yml

```diff
@@ -27,8 +27,28 @@ jobs:
         with:
           ref: ${{ github.event.pull_request.head.sha }}
 
+      - name: Get changed files
+        id: changed-files-specific
+        uses: tj-actions/changed-files@v44
+        with:
+          files_ignore: |
+            release-notes/*.md
+            .github/**
+            ADMINS.md
+            CHANGELOG-3.0.md
+            CHANGELOG.md
+            CODE_OF_CONDUCT.md
+            CONTRIBUTING.md
+            DEVELOPER_GUIDE.md
+            MAINTAINERS.md
+            README.md
+            RELEASING.md
+            SECURITY.md
+            TESTING.md
+            TRIAGING.md
+
       - name: Setup environment variables (PR)
-        if: github.event_name == 'pull_request_target'
+        if: github.event_name == 'pull_request_target' && steps.changed-files-specific.outputs.any_changed == 'true'
```

Comment:
Can we split out the gradle check setup and workflow into a separate job, and make the whole job dependent?

Reply:
See https://github.com/opensearch-project/OpenSearch/pull/13498#discussion_r1586986521

Reply:
:+1: to @dblock 's suggestion, the same condition is propagated to (mostly) every step

Reply:
So I did try that but does not work as expected. See comment posted above. Let me know if you can think of better way.

Reply:
I don't think I understand why the gradle check job is different from other jobs. It should be skippable, shouldn't it?

Reply:
The issue is this repository has added `gradle-check` as a required workflow check before merging any PR. GitHub is unable to resolve the conflict if the workflow is skipped due to any condition. It HAS to be passed. Not failed, not skipped :(

Reply:
@gaiksaya I think there might be another approach as described here: https://stackoverflow.com/questions/65844033/conditional-needs-in-github-actions

We'd need to something like:

```
jobs:
  changed-files-specific:
    steps:
      - set the any_changed boolean

  gradle-check:
    runs-on: ubuntu-latest
    needs: changed-files-specific
    if: ${{ needs.changed-files-specific.outputs.any_changed == 'true' }}
    steps:
      - run the gradle check steps

  check-result:
    needs: gradle-check
    if: always()
    steps:
      - name: fail if conditional job failed
        if: ${{ needs.changed-files-specific.outputs.any_changed == 'true' && needs.gradle-check.result == 'failure' }}
        run: exit 1
```

And then the branch protection rule would be on `check-result` which will always run but is smart enough to ignore gradle-check if no relevant files changed. Would this work?

Reply:
Sorry for delayed response. Missed this. Not really, it has to pass if it is required. Cannot skip or fail. So I believe only approach I think is follow what you just posted and add one more step which will print something like "Skipping check" and show it as pass. Let me know if everyone are okay with this approach

Reply:
@gaiksaya In my suggestion I think the `check-result` step will always run, and it is the thing that will be marked as required. Why would you need another step?

Reply:
I think you are right. @reta @dblock Are you okay with this approach? I can go ahead and test it out and move to implementation then.

Reply:
👍 thanks @gaiksaya !

Reply:
Yes i am, thank you.

Reply:
Updated with code with suggested changes. Thanks! Also works as expected. Here is the sample: https://github.com/gaiksaya/OpenSearch/pull/12